src/plugins/ui_actions/tsconfig.json

```diff
@@ -0,0 +1,17 @@
+{
+  "extends": "../../../tsconfig.base.json",
+  "compilerOptions": {
+    "composite": true,
+    "outDir": "./target/types",
+    "emitDeclarationOnly": true,
+    "declaration": true,
+    "declarationMap": true
+  },
+  "include": ["public/**/*"],
+  "references": [
+    { "path": "../../core/tsconfig.json" },
+    { "path": "../expressions/tsconfig.json" },
+    { "path": "../kibana_utils/tsconfig.json" },
+    { "path": "../kibana_react/tsconfig.json" },
```

Comment:
I see you changed the `expressions` import to be `import type`, but I wonder if we should add `expressions` as a `requiredBundle` in `ui_action`'s `kibana.json` file? I'm afraid of these inconsistencies between `tsconfig.json` and `kibana.json`, wdyt? Theorically, a plugin shouldn't import code from a plugin it don't have dependency on, right?

Reply:
It's not an obligatory requirement since `requiredBundle` is for runtime deps. I might add it to `requiredBundle`, but there is no way to enforce it. One way to do so is to generate `tsconfig.json` file from `kibana.json`. @spalger WDYT?

Reply:
We verify that people don't have unnecessary `requiredBundles` by ensuring that each is actually used, but we aren't able to do that if we start including type-only dependencies in the list. I think it's better to keep the `requiredBundles` array as small as possible and that includes removing type-only dependencies as well as bundles which are no longer required because usage was removed.